### ISSUE TITLE
x360ce does not display all input (including virtual/ViGEmBus) devices

### ISSUE THREAD

**Issue by JayMontana36** [x360ce/x360ce#1142]
As the title states, x360ce does not detect or display virtual/ViGEmBus devices, which is the source of most if not all issues relating to DS4Win+x360ce or wireless/Bluetooth PS4/DS4 or Bluetooth controllers, etc.
** **
Now somehow, I managed in my previous installation of Windows to select virtual ViGEmBus devices created by DS4Win with an older version of x360ce (4.x) that I migrated to just last month (from 3.x) that I have since lost after reinstalling Windows not too long ago, but it was a 4.x build from May 2019, I'll see if I can find/recover it and try it again. It would be nice to set up my x360ce virtual Xbox 360 controller in the following manner (again):

 - DS4/PS4 controller connected over Bluetooth with communications to/from the controller managed/handled completely/entirely by DS4Win (x360ce has nothing at all to do with (and does not care about) this portion)

- DS4Win is also configured to create a virtual "wireless controller" aka virtual PS4/DS4 controller (instead of a virtual Xbox 360 controller) through ViGEmBus, which happens to be seen by x360ce in the same exact way as if the controller were connected via USB instead of Bluetooth (and, Windows does also even show/display the inputs from the physical controller through the virtual controller in "Game Controllers", just like it would if/when the controller is connected via USB instead of Bluetooth), and is also configured to hide the physical input from showing up at all so that only the virtual output is visible, which is all that x360ce should need.

 - x360ce is configured to take the outputs from the virtual ViGEmBus DS4/PS4 controller as inputs for it's virtual Xbox 360 controller's outputs, alongside any other game controllers (flight sticks, throttles, racing/steering wheels, etc.) virtual or physical, in whatever x360ce configuration you'd like (which for me would be a single combined Xbox 360 controller output, consisting of my DS4Win's virtual DS4 controller outputs alongside my physical flight/joy stick's (and/or whatever other controllers exist out there) inputs/outputs (with some (anti)deadzone and sensitivity tweaks through x360ce for my flight/joy stick of course), probably also hiding the sources of the inputs for x360ce's controller outputs from Windows and applications.

 - I hope I made sense here without overdoing it; probably not though.
** **
**tl;dr**: it would be nice to have an option to show (and not hide) virtual ViGEmBus devices in x360ce so that things like DS4Win will work with x360ce (again), allowing us to use our DS4/PS4 (or whatever other) controllers over Bluetooth with x360ce, since the purpose of x360ce is to take an input and (re)map it to Xbox 360 game pad Xinput, regardless of whether that input is physical or virtual.

**Comment by JayMontana36** [comment]
Well, apparently I somehow got it to work again (don't know what I did or what happened, simply launched DS4Win then x360ce and somehow it worked, originally showing both as connected via USB then changed the DS4 connected via icon to a PC?)

![image](https://user-images.githubusercontent.com/6281870/89583326-20367800-d808-11ea-822e-76d9998f2471.png)
![image](https://user-images.githubusercontent.com/6281870/89583929-3264e600-d809-11ea-9975-511e3f928e7d.png)

Well still, it's difficult to get this working correctly/properly initially for no reason, any caveats I come across while testing I will post here. But this should serve as a micro-guide on how to get this to work correctly for those who don't know.

**Comment by EJocys** [comment]
Thank you for reporting. I will check what I can do in next releases. Showing and hiding controllers is next on TODO list.

**Comment by SoggieWafflz** [comment]
yeah not getting this to work at all with either DS4Windows or DSX

**Comment by EvoJocys** [comment]
The program's own emulated controllers are its output, not an input, so they are not offered as devices to map; since 4.20.43 the device list names the XInput place each device reaches, including through the controllers it is mapped to, and leftover virtual controllers are listed on the Issues tab.

Closing as part of a tracker clean-up after the 4.21 release.
